File: dart/lib/bebop_handlers/new_game.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class NewGameMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = NewGameMsg.decode(message);
    GameState.instance.setResetCounter();
    GameState.instance.signalNewGame(message);
    print('Starting new game');
  }
}

File: typescript/src/bebop_handlers/newGame.ts
import GameState from '../classes/gameState';
import { NewGameMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class NewGameMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = NewGameMsg.decode(message);
        GameState.Instance.setResetCount();
        GameState.Instance.signalNewGame(message);
        console.log('Starting new game');
    }
}